### PR TITLE
Find/Replace overlay: disable search scope when leaving #1919

### LINKED ISSUE #1919: [find/replace overlay] leaving overlay doesnt remove search scope
When selecting a search scope and leaving the Overlay, the search scope is not reset as it should be and as it is currently in the find/replace dialog.

### PR DESCRIPTION
**Pull request by Wittmaxi**
Replace the FocusListener with a ShellAdapter which will remove the
search scope as soon as the overlay is left.
The FocusListener didn't listen to the case where the focus is in the
search bar and the search bar is left (and the overlay is left
implicitly)

fixes https://github.com/eclipse-platform/eclipse.platform.ui/issues/1919

### CODE REVIEW

**Comment by Wittmaxi** [comment]
Random failures are not related to this PR and documented in https://github.com/eclipse-platform/eclipse.platform.ui/issues/751

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java [hunk 1]
```diff
@@ -226,11 +228,22 @@ public void controlResized(ControlEvent e) {
 		}
 	};
 
-	private FocusListener overlayFocusListener = FocusListener.focusLostAdapter(e -> {
-			findReplaceLogic.activate(SearchOptions.GLOBAL);
-			searchInSelectionButton.setSelection(false);
-	});
+	/**
+	 * Updates the find replace dialog on activation changes.
```
Comment (HeikoKlare) [inline bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java]: This listener is about the "overlay", not the "dialog". I think the comment is even obsolete, as an activation listener obviously listens to activation changes, so I think you can remove the comment.

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java [hunk 2]
```diff
@@ -226,11 +228,22 @@ public void controlResized(ControlEvent e) {
...
+			// Do nothing
+		}
 
+		@Override
+		public void shellDeactivated(ShellEvent e) {
+			removeSearchScope();
+		}
+	}
+	
+	private ActivationListener overlayFocusListener = new ActivationListener();
```
Comment (HeikoKlare) [inline bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java]: You could also assign an anonymous class with the according implementation instead of defining a new `ActivationListener` class, i.e., `private ShellListener overlayDeactivationListener = new ShellAdapter() { ...};
```suggestion
	private ActivationListener overlayDeactivationListener = new ActivationListener();
```

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlayFirstTimePopup.java [hunk 3]
```diff
@@ -96,7 +96,6 @@ public static void displayPopupIfNotAlreadyShown(Shell shellToUse) {
 							true)
 					.delay(POPUP_VANISH_TIME.toMillis()).open();
 		}
-
```
Comment (HeikoKlare) [inline bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlayFirstTimePopup.java]: Please revert this unnecessary formatting change.

**Review by HeikoKlare — COMMENTED** [review]
Works as expected. Only some minor comments on code style / naming / documentation.

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java (hunk 1 above)
Comment (Wittmaxi) [reply bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java]: Yes, I agree, the comment is not helpful and even out of date. Removed: https://github.com/eclipse-platform/eclipse.platform.ui/commit/44181631466acf7bcceca1f4d7ae6ab55327c1d0

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java (hunk 2 above)
Comment (Wittmaxi) [reply bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java]: Good points, I agree that they improve the code quality!

Addressed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/f4a575c53678d57f08428e30573dc666f79cebbd

File: bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlayFirstTimePopup.java (hunk 3 above)
Comment (Wittmaxi) [reply bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlayFirstTimePopup.java]: Yes! https://github.com/eclipse-platform/eclipse.platform.ui/commit/f4a575c53678d57f08428e30573dc666f79cebbd

**Review by HeikoKlare — APPROVED** [review]
Great. This looks good now. Let's wait for the build results and I expect that we can merge then.
